File: backend/src/main/java/com/campus/learningspace/entity/Library.java
package com.campus.learningspace.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("library")
public class Library {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String name;

    private String address;

    private Integer floorCount;

    private String description;

    private BigDecimal latitude;

    private BigDecimal longitude;

    private String openingHours;

    private Integer status;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @TableLogic
    private Integer deleted;
}

File: frontend/src/api/library.ts
import request from '@/utils/request'

export interface Library {
  id?: number
  name: string
  address?: string
  floorCount?: number
  description?: string
  latitude?: number
  longitude?: number
  openingHours?: string
  status?: number
}

export interface Result<T> {
  code: number
  message: string
  data: T
}

export const getLibraryList = () => {
  return request.get<any, Result<Library[]>>('/library/list')
}

